File: app/src/main/java/se/hactar/movieregister/helper/imdb/model/Result.java
package se.hactar.movieregister.helper.imdb.model;


import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public final class Result {
    @SerializedName("l")
    public String title;

    @SerializedName("id")
    public String id;

    @SerializedName("s")
    public String starring;

    @SerializedName("y")
    public int year;

    @SerializedName("q")
    public String type;

    @SerializedName("i")
    public final List<String> image = new ArrayList<>();

    public String getTitle() {
        return title == null ? "" : title;
    }

    public String getId() {
        return id == null ? "" : id;
    }

    public String getStarring() {
        return starring == null ? "" : starring;
    }

    public int getYear() {
        return year;
    }

    public String getType() {
        return type == null ? "" : type;
    }

    public String getImageUrl() {
        if (!image.isEmpty()) {
            return image.get(0);
        }
        return "";
    }

    @Override
    public String toString() {
        return "Result{" +
                "title='" + title + '\'' +
                ", id='" + id + '\'' +
                ", starring='" + starring + '\'' +
                ", year=" + year +
                ", type='" + type + '\'' +
                ", imageUrl=" + getImageUrl() +
                '}';
    }
}

File: app/src/main/java/se/hactar/movieregister/helper/imdb/model/Result.kt
package se.hactar.movieregister.helper.imdb.model


import com.google.gson.annotations.SerializedName

import java.util.ArrayList

class Result {
    @SerializedName("l")
    var _title: String? = null
    val title: String
        get() = _title ?: ""

    @SerializedName("id")
    private var _id: String? = null
    val id: String
        get() = _id ?: ""

    @SerializedName("s")
    private var _starring: String? = null
    val starring: String
        get() = _starring ?: ""

    @SerializedName("y")
    var year: Int = 0

    @SerializedName("q")
    private var _type: String? = null
    val type: String
        get() = _type ?: ""

    @SerializedName("i")
    private var image: List<String> = ArrayList()

    val imageUrl: String
        get() = if (!image.isEmpty()) image[0] else ""

    override fun toString(): String {
        return "Result{" +
                "title='" + title + '\''.toString() +
                ", id='" + id + '\''.toString() +
                ", starring='" + starring + '\''.toString() +
                ", year=" + year +
                ", type='" + type + '\''.toString() +
                ", imageUrl=" + imageUrl +
                '}'.toString()
    }
}
